exporter/datadogexporter/internal/hostmetadata/metadata.go

```diff
@@ -91,8 +92,25 @@ func fillHostMetadata(params exporter.CreateSettings, pcfg PusherConfig, p sourc
 
 func (p *pusher) pushMetadata(hm payload.HostMetadata) error {
 	path := p.pcfg.MetricsEndpoint + "/intake"
-	buf, _ := json.Marshal(hm)
-	req, _ := http.NewRequest(http.MethodPost, path, bytes.NewBuffer(buf))
+	marshaled, err := json.Marshal(hm)
```

Comment:
where was/is the content-encoding header set?

Reply:
II added a comment on f9f05b7 to make it clear. It's set on the line below the new comment, see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4f75af758f343344a72ed3d5bcc34cbe56e2283e/exporter/datadogexporter/internal/clientutil/http.go#L19-L22